### PR TITLE
perf: better cast create2

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Speed

## Solution

Make faster

### CODE REVIEW

File: crates/cast/bin/cmd/create2.rs [hunk 1]
```diff
@@ -114,34 +124,62 @@ impl Create2Args {
...
+            let increment = n_threads;
+            let start = i;
+            let regex = regex.clone();
+            let regex_len = regex.patterns().len();
+            let found = Arc::clone(&found);
+            handles.push(std::thread::spawn(move || {
+                struct B256Aligned(B256, [usize; 0]);
+                let mut salt = B256Aligned(B256::ZERO, []);
+                let salt_word = unsafe { &mut *salt.0.as_mut_ptr().cast::<usize>() };
+                *salt_word = start;
```
Comment (mattsse) [inline crates/cast/bin/cmd/create2.rs:149]: not obvious what's going on here, this sets the salt's first word to the thread's index incrementing it by the number of threads, so the salt cycles through all unique values?

needs docs+SAFETY

**Review by mattsse — COMMENTED** [review]
speed gud

**Review by Evalir — APPROVED** [review]
haha create2 go brrrr

same as matt, just more docs would be nice!

File: crates/cast/bin/cmd/create2.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/cast/bin/cmd/create2.rs:149]: n_threads=4
- word=0; word=4; word=8
- word=1; word=5; word=9
- word=2; word=6; word=10
- word=3; word=7; word=11

**Comment by DaniPopes** [comment]
Added

File: crates/cast/bin/cmd/create2.rs [hunk 2]
```diff
@@ -136,36 +136,49 @@ impl Create2Args {
...
             let found = Arc::clone(&found);
             handles.push(std::thread::spawn(move || {
+                // Read the first bytes of the salt as a usize to be able to increment it.
                 struct B256Aligned(B256, [usize; 0]);
                 let mut salt = B256Aligned(B256::ZERO, []);
+                // SAFETY: B256 is aligned to `usize`.
                 let salt_word = unsafe { &mut *salt.0.as_mut_ptr().cast::<usize>() };
-                *salt_word = start;
+
+                // Important: set the salt to the start value, otherwise all threads loop over the
```
Comment (mattsse) [inline crates/cast/bin/cmd/create2.rs:155]: the loop only covers u64 or u32 depending on the arch, so not all the available 32bytes?

fine?

Comment (DaniPopes) [reply crates/cast/bin/cmd/create2.rs:155]:
```
>> u32::MAX / 128
33554431
>> u64::MAX / 128
144115188075855871
```

iterations needed to overflow, I don't think that's reachable

**Review by mattsse — APPROVED** [review]

**Review by Evalir — APPROVED** [review]

**Pull request merged**